File: rb/features.rb
# CollegeRoi SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module CollegeRoiFeatures
  def self.make_feature(name)
    case name
    when "base"
      CollegeRoiBaseFeature.new
    when "test"
      CollegeRoiTestFeature.new
    else
      CollegeRoiBaseFeature.new
    end
  end
end

File: py/features.py
# CollegeRoi SDK feature factory

from feature.base_feature import CollegeRoiBaseFeature
from feature.test_feature import CollegeRoiTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CollegeRoiBaseFeature(),
        "test": lambda: CollegeRoiTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
